src/email.c

```diff
@@ -34,66 +34,123 @@ compose_mail_mailto (GAppInfo       *info,
 {
   g_autofree char *enc_subject = NULL;
   g_autofree char *enc_body = NULL;
+  g_autofree char *arg = NULL;
   g_autoptr(GString) url = NULL;
   g_autoptr(GList) uris = NULL;
+  g_autoptr(GString) tb_arg = NULL;
   int i;
   gboolean success;
   const char *sep;
+  const char *is_tb_snap;
+  const char *old_cmdline;
 
   enc_subject = g_uri_escape_string (subject ? subject : "", NULL, FALSE);
   enc_body = g_uri_escape_string (body ? body : "", NULL, FALSE);
 
-  url = g_string_new ("mailto:");
-
-  sep = "?";
-
-  if (addrs)
-    {
-    for (i = 0; addrs[i]; i++)
-      {
-        if (i > 0)
-          g_string_append (url, ",");
-        g_string_append_printf (url, "%s", addrs[i]);
+  /* The commandline can be complicated, e.g.
+   * env BAMF_DESKTOP_FILE_HINT=/var/lib/snapd/.../thunderbird_thunderbird.desktop /snap/bin/thunderbird %u
+   * so just match for thunderbird.
+   */
+  old_cmdline = g_app_info_get_commandline(info);
+  if(strstr(old_cmdline, "thunderbird")) {
```

Comment:
Can't use `g_app_info_get_executable`?

Reply:
The problem is that in the example command line commented above, that would be `env`.